specification/azureactivedirectory/resource-manager/Microsoft.Aadiam/preview/2017-04-01-preview/azureactivedirectory.json

```diff
@@ -0,0 +1,543 @@
+  "info": {
+    "version": "2017-04-01-preview",
+    "title": "azureactivedirectory",
+    "description": "Azure Active Directory Client.",
+    "x-ms-code-generation-settings": {
+      "name": "azureactivedirectoryClient"
+    }
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/microsoft.aadiam/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "Operations_List",
+        "x-ms-examples": {
+          "Gets the operations": {
+            "$ref": "./examples/listOperations.json"
+          }
+        },
+        "description": "Operation to return the list of available operations.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get list of AadIam operations.",
+            "schema": {
+              "$ref": "#/definitions/OperationsDiscoveryCollection"
+            }
+          },
+          "default": {
+            "description": "Error, failed to get list of AadIam operations.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/microsoft.aadiam/diagnosticSettings": {
+      "get": {
+        "tags": [
+          "DiagnosticSettings"
+        ],
+        "operationId": "DiagnosticSettings_List",
+        "description": "Gets the active diagnostic settings list for AadIam.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. Request for AadIam diagnostic settings has failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to get more information about AadIam diagnostic settings",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticSettingsResourceCollection"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic settings": {
+            "$ref": "./examples/listDiagnosticSettings.json"
+          }
+        }
+      }
+    },
+    "/providers/microsoft.aadiam/diagnosticSettings/{name}": {
```

Comment:
>microsoft.aadiam/diagnosticSettings [](http://example.com/codeflow?start=16&length=35)

DiagnosticSettings typs is already there inside Microsoft.Insights namespace. Please explain why this object is being added?
Also why are we updated 4 year old API and not creating new version. It's not allowed to update older than 1 year API!

Reply:
I'm not sure how else to add this endpoint. Ours is for microsoft.aadiam. This is how it's been done in previous pull requests, is there a way to cover microsoft.aadiam another way?

This is an s360 requirement, 2017-04-01-preview endpoints are currently being used and s360 is asking our team to add docs for it since none currently exist for that version. These docs have no changes from the stable 2017-04-01 spec.